internal/prober/scaler/scale.go

```diff
@@ -25,6 +25,8 @@ const (
 	// replicasAnnotationKey is the key for an annotation whose value captures the current spec.replicas prior to scale down for that resource.
 	// This is used when DWD attempts to restore the state of the resource it scale down.
 	replicasAnnotationKey = "dependency-watchdog.gardener.cloud/replicas"
+	// activeAnnotation captures if the DWD is active on a resource so that shoot reconciliation can ignore this resource from scaling up.
+	scaledDownAnnotationKey = "dependency-watchdog.gardener.cloud/scaled-down"
```

Comment:
Is it not sufficient to indicate that DWD is active on a specific resource? Do we need to specify that the operation performed by DWD was `scaled-down`?
Perhaps we can use a more semantic name like `dependency-watchdog.gardener.cloud/meltdown-protection-active`. WDYT?

Reply:
Ok my thinking was that once you add an annotation indicating that dwd is currently scaling this resource, then both scale-in and scale-out operations are automatically covered. Gardener looks at this annotation and knows that during its reconciliation it should not scale this resource yet since DWD is acting on this resource.

Also since this is now going to be used by gardener as well, it therefore becomes part of the API. Can you move this annotation to the API (api/prober/types.go) - there you can define a constant for this annotation and add proper doc comments.

Reply:
As suggested, adapted via https://github.com/gardener/dependency-watchdog/pull/140/commits/1c3a55ae5bc69bb9e9171e51bc0c017a4ae30fc9.